k2/csrc/cuda/arch.h

```diff
@@ -0,0 +1,32 @@
+// k2/csrc/cuda/arch.h
+
+// Copyright (c) 2020 Xiaomi Corporation ( authors: Meixu Song )
+
+// See ../../LICENSE for clarification regarding multiple authors
+#ifndef K2_CSRC_CUDA_ARCH_H_
+#define K2_CSRC_CUDA_ARCH_H_
+
+namespace k2 {
+
+/**
+ * @def K2_PTX_ARCH
+ *
+ * @brief
+ *  K2_PTX_ARCH get the target PTX version through the active compiler.
+ *
+ * @details
+ *  - for host, it's 0.
+ *  - for device, it's `__CUDA_ARCH__`, which indicates the
+ *                compute compatibility, should >= 200.
+ */
+#ifndef K2_PTX_ARCH
```

Comment:
You have macro guards around each definition of a macro.  Can't you just use an include guard around the entire header?  Wouldn't that be simpler?

Reply:
The include guard is already there. The intension is to avoid conflict with following redefine the same name macro outer the header. Though it rarely happens, sometimes could, e.g. conflict with `K2_PARANOID_ASSERT` from kaldi-error.h, `LOG()` from glog.

Reply:
Macros that start with K2_ don't need include guards as they're specific to this project.